runtime/caches/tiered.ts

```diff
@@ -0,0 +1,141 @@
+import { logger } from "../../observability/otel/config.ts";
+
+const inFuture = (maybeDate: string) => {
+  try {
+    return new Date(maybeDate) > new Date();
+  } catch {
+    return false;
+  }
+};
+const isCache = (c: Cache | undefined): c is Cache => typeof c !== "undefined";
+
+export function createTieredCache(
```

Comment:
Could you add a JSDoc of this function? just explaining the overall behavior

Reply:
Will do